src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1238,13 +1240,13 @@ private static TokenValidationResult ReadToken(string token, TokenValidationPara
             JsonWebToken jsonWebToken = null;
             if (validationParameters.TokenReader != null)
             {
-                var securityToken = validationParameters.TokenReader(token, validationParameters);
+                var securityToken = validationParameters.TokenReader(token, validationParameters); 
                 if (securityToken == null)
-                    throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10510, token)));
+                    throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10510, LogHelper.MarkAsSecurityArtifact(token))));
```

Comment:
>MarkAsSecurityArtifact [](http://example.com/codeflow?start=162&length=22)

This method wraps the argument in a SecurityArtifact object, yet this type is not used anywhere on the LogHelper. Since the current implementation relies on the ISafeLogSecurityArtifact interface there is no need for this helper method and the SecurityArtifact type.

Reply:
We only implement the interface for the object derived from SecurityToken. Here we log the token as string that the LogHelper cannot scrub the signature for a string.